app/controllers/sales_controller.rb

```diff
@@ -0,0 +1,64 @@
+class SalesController < ApplicationController
+  def index
+    @vendor = Vendor.find(params[:vendor_id])
+    @sales = @vendor.recent_sales_first
+
+    if @sales.length == 0
+      render :empty
```

Comment:
This works, but I typically see conditionals in the view rather than rendering an entirely new view.